### PR TITLE
Implement partial line group update

### PR DESCRIPTION
**Pull request by vangberg**
## Summary

Fixes the issue where partial execution (Cmd+Enter on a single line) would clear all previously executed results. Now preserves line groups that don't overlap with the executed range.

**Before:** Executing line 3 would remove results from lines 1, 2, 4, etc.  
**After:** Executing line 3 only updates line 3's results, keeping all others intact.

## Changes

- Updated `processExecutionResults()` in `src/results.ts` to accept `lineRange` and `currentLineGroups` parameters
- Implemented overlap detection logic: `group.lineEnd < from || group.lineStart > to`
- Non-overlapping groups are preserved, overlapping groups are replaced
- Threaded `lineRange` through `App.tsx` execution flow to the results hook
- Added comprehensive test suite with 8 test cases covering edge cases
- Included plan document `plans/2025-011-partial-line-group-update.md`

## Test plan

- [x] Run all existing tests (36 tests pass)
- [x] Test with Chrome MCP: execute full script, then partial execution on single line
- [x] Verify non-overlapping groups preserved
- [x] Verify overlapping groups replaced correctly
- [x] Test multiple partial executions in sequence

### CODE REVIEW

File: src/results.ts [hunk 1]
```diff
@@ -18,18 +18,43 @@ export function addResultsToStore(
...
  * Processes execution results: adds to store and computes line groups.
+ * For partial execution, merges new groups with non-overlapping existing groups.
  */
 export function processExecutionResults(
   resultStore: Map<number, ExecutionOutput>,
-  newResults: ExecutionOutput[]
+  newResults: ExecutionOutput[],
+  options?: {
+    currentLineGroups?: LineGroup[];
+    lineRange?: { from: number; to: number };
```
Comment (vangberg) [inline src/results.ts:28]: Can't we just infer this from `newResults`?
